ci/build.py

```diff
@@ -58,7 +60,11 @@ def get_docker_tag(platform: str, registry: str) -> str:
     platform = platform if any(x in platform for x in ['build.', 'publish.']) else 'build.{}'.format(platform)
     if not registry:
         registry = "mxnet_local"
-    return "{0}/{1}".format(registry, platform)
+    dockerfile_hash = 'unk'
+    with open(get_dockerfile(platform),"rb") as f:
+        bytes = f.read()
+        dockerfile_hash = hashlib.sha256(bytes).hexdigest()[:12]
```

Comment:
Just hashing the dockerfile is not sufficient IMO. It would cause a lot of collisions. Instead, I'd recommend to hash the entire Docker buildcontext directory. 

The impact of these collisions would be that different jobs with actually different buildcontexts would constantly override the cache and thus basically invalidate it. 

When you are creating the hash out of the entire context, please make sure to use a chunk/streaming based hash calculation instead of reading everything into the ram for efficiencys sake.

Reply:
Also, be aware that this method will result in a 100% cache miss once anything under that hash changed since there will simply be no image with that particular hash. Thus I'd recommend to write a fallback method which first tries to pull the hashed cache and in case it does not exist, pull the branch cache - which obviously requires images pushed by the branch jobs to be appropriately tagged as such.

Reply:
Updated to hash all files copied into container.